bugtool/configuration/configuration.go

```diff
@@ -0,0 +1,379 @@
+		return nil
+	}
+	commands := []string{}
+	for _, iface := range ifaces {
+		commands = append(commands,
+			fmt.Sprintf("tc filter show dev %s ingress", iface.Name),
+			fmt.Sprintf("tc filter show dev %s egress", iface.Name),
+			fmt.Sprintf("tc chain show dev %s", iface.Name),
+			fmt.Sprintf("tc class show dev %s", iface.Name),
+		)
+	}
+	return commands
+}
+
+func systemFileDumps() []dump.Task {
+	ts := []dump.Task{}
+	files := []string{
+		"/proc/net/xfrm_stat",
+		"/proc/sys/net/core/bpf_jit_enable",
+		"/proc/kallsyms",
+		"/etc/resolv.conf",
+		"/var/log/docker.log",
+		"/var/log/daemon.log",
+		"/var/log/messages",
+	}
+	for _, f := range files {
+		ts = append(ts, dump.NewFile(f))
+	}
+	return ts
+}
+
+// routeCommands gets the routes tables dynamically.
+func routeCommands() []dump.Task {
+	// oneline script gets table names for all devices, then dumps either ip4/ip6 route tables.
+	routesScript := "for table in $(ip --json %s route show table all | jq -r '.[] | select(.table != null) | select(.table != \"local\") | .table'); do ip --json %s route show table $table ; done"
+	var commands []dump.Task
+	commands = append(commands,
+		dump.NewExec("ip4-route-tables", "json", "bash", []string{"-c", fmt.Sprintf(routesScript, "-4", "-4")}...),
+		dump.NewExec("ip6-route-tables", "json", "bash", []string{"-c", fmt.Sprintf(routesScript, "-6", "-6")}...),
+	)
+	return commands
+}
+
+// Returns commands that have bespoke output formatting, designed
+// for human readability over machine parsing.
+// Note: These are deprecated and are here for legacy reasons.
+// Avoid adding commands that cannot be easily parsed by a machine
+// (preferably in json).
+// If necessary, it may be preferable to write functionality as a bugtool/dump.Task.
+func humanReadableCommands() []string {
+	return []string{
+		"top -b -n 1",
+		"uptime",
+
+		// ss
+		"ss -H -t -p -a -i -s",
+
+		// ps
+		"ps auxfw", // todo: rework this, use go, add ppid
+
+		// iptables
+		"iptables-save -c",
+		"ip6tables-save -c",
+		"iptables-nft-save -c",
+		"ip6tables-nft-save -c",
+		"iptables-legacy-save -c",
+		"ip6tables-legacy-save -c",
+		"ipset list",
+	}
+}
+
+func logCommands() []string {
+	return []string{
+		"dmesg --time-format=iso",
+	}
+}
+
+func tableStructuredCommands() []string {
+	return []string{
+		// Host and misc
+		"hostname",
+		"uname -a",
+		"uptime",
+		"sysctl -a",
+		"taskset -pc 1",
+		"lsmod",
+
+		"ss -H -u -p -a -s",
+	}
+}
+
+func tcCommands() dump.Tasks {
+	ts := dump.Tasks{}
+	for _, c := range []string{
+		// tc
+		"tc -j -s qdisc show", // Show statistics on queuing disciplines
+	} {
+		ts = append(ts, createExecFromString(c, "json"))
+	}
+	return ts
+}
+
+// Contains commands that output json.
+func jsonStructuredCommands() dump.Tasks {
+	ts := dump.Tasks{}
+	for _, c := range []string{
+		// ip
+		"-j a",
+		"-j -4 r",
+		"-j -6 r",
+		"-j -d -s l",
+		"-j -4 n",
+		"-j -6 n",
```

Comment:
I'm not finding the output for these two but I have a `ip-n.json` file with a broken JSON content:
```json
[]
dst":"10.164.0.1","dev":"eth0","lladdr":"42:01:0a:a4:00:01","state":["REACHABLE"]},{"dst":"10.68.1.24","dev":"lxc4eac24edaf5e","lladdr":"a6:0a:28:ec:2b:ad","state":["REACHABLE"]},{"dst":"10.68.1.49","dev":"lxc_health","lladdr":"ee:b0:b4:e3:84:67","state":["DELAY"]},{"dst":"10.68.1.249","dev":"lxcaa96ecd074ee","lladdr":"ba:4a:5e:44:da:24","state":["REACHABLE"]},{"dst":"10.68.1.133","dev":"lxcb94237c3f348","lladdr":"de:a4:de:da:28:d4","state":["REACHABLE"]},{"dst":"10.68.1.193","dev":"lxcad1b39b5fa79","lladdr":"f6:ed:6b:7d:a1:04","state":["REACHABLE"]}]

```

Reply:
Thats odd, could be a potential bug in the ip command. The json encoding it uses seems potentially error prone (was taking a look at why xfrm json output doesn't work).

Reply:
Although I'm surprised it's different, considering the tool is part of the container image. I have no problems with it:

```
[{"dst":"fe80::149f:d2ff:fec5:5a74","dev":"lxc_health","lladdr":"16:9f:d2:c5:5a:74","state":["STALE"]},{"dst":"fc00:c111::3","dev":"eth0","lladdr":"02:42:ac:13:00:03","router":null,"state":["REACHABLE"]}]}]
```

Did you build the image using the dev image targets?